e2e/e2ek6.test.js

```diff
@@ -0,0 +1,307 @@
+import { chromium } from 'k6/experimental/browser';
+import { check, fail } from 'k6';
+import http from 'k6/http';
+
+import { URL } from 'https://jslib.k6.io/url/1.0.0/index.js';
+import { uuidv4 } from 'https://jslib.k6.io/k6-utils/1.4.0/index.js';
+import { selectors } from 'https://unpkg.com/@grafana/e2e-selectors/dist/index.js';
+
+const DASHBOARD_TITLE = `e2e-test-dashboard-${uuidv4()}`;
+const DATASOURCE_NAME = `ClickHouse-e2e-test-${uuidv4()}`;
+let orgID;
+let datasourceUID;
+let apiToken;
+const getDashboardUid = (url) => {
+  const matches = new URL(url).pathname.match(/\/d\/([^/]+)/);
+  if (!matches) {
+    throw new Error(`Couldn't parse uid from ${url}`);
+  } else {
+    return matches[1];
+  }
+};
+
+export async function login(page) {
+  try {
+    const loginURL = selectors.pages.Login.url;
+    await page.goto(`http://localhost:3000${loginURL}`, { waitUntil: 'networkidle' });
+
+    const usernameInput = page.locator(`input[aria-label="${selectors.pages.Login.username}"]`)
+    await usernameInput.type('admin');
+    const passwordInput = page.locator(`input[aria-label="${selectors.pages.Login.password}"]`)
+    await passwordInput.type('admin');
+    const submitButton = page.locator(`button[aria-label="${selectors.pages.Login.submit}"]`)
+    await submitButton.click();
+
+    // checks page for skip change password screen
+    check(page, {
+      'change password is presented':
+        page.locator(`button[aria-label="${selectors.pages.Login.skip}"]`).textContent() === 'Skip',
+    });
+  } catch (e) {
+    fail(`login failed: ${e}`);
+  } 
+};
+
+export async function addDatasource(page) {
+  try {
+    const addDataSourceURL = selectors.pages.AddDataSource.url;
+    await page.goto(`http://localhost:3000${addDataSourceURL}`, { waitUntil: 'networkidle' });
+    
+    const clickHouseDataSource = page.locator(`button[aria-label="${selectors.pages.AddDataSource.dataSourcePluginsV2('ClickHouse')}"]`);
+    await clickHouseDataSource.click();
+    const dataSourceName = page.locator(`input[aria-label="${selectors.pages.DataSource.name}"]`);
+    dataSourceName.fill('');
+    dataSourceName.type(`${DATASOURCE_NAME}`);
+    const serverAddress = page.locator(`input[aria-label="Server address"]`);
+    serverAddress.type('localhost');
+    const serverPort = page.locator('input[aria-label="Server port"]');
+    serverPort.type('9000');
+    const saveAndTestButton = page.locator(`button[data-testid="data-testid ${selectors.pages.DataSource.saveAndTest}"]`);
+    await saveAndTestButton.click();
+
+    // checks the page for the data source is working message
+    check(page, {
+      'add datasource successful':
+      await page.locator('[aria-label="Create a dashboard').textContent() === "building a dashboard",
+    });
+
+    const orgName = `api-org-${uuidv4()}`
+
+    // creates org
+    const getOrg = http.post('http://admin:admin@localhost:3000/api/orgs', `{"name": "${orgName}"}`, {
```

Comment:
Why do these steps need to be carried out if the save and test flow succeeds? Are you effectively adding the datasource twice?

Reply:
You are correct re:adding the datasource twice - good catch. The idea was to check the status code of `checkHealth` not just the success message. Also by creating a datasource this way it was giving access to the `datasouceUID` which was allowing a query to be ran in a later test. I think what makes most sense now is to remove this flow and keep the first one (typing in localhost/9000, etc.) I will amend the `configurePanel` test to get the UID another way.